### PR TITLE
[subset] Add hdmx subsetting.

### PR DESCRIPTION
**Pull request by garretrieger**

### CODE REVIEW

**Comment by behdad** [comment]
Curious. Why hdmx? No one uses that...

File: src/hb-ot-hdmx-table.hh [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  HBUINT16 version;
+  HBINT16  num_records;
+  HBINT32  size_device_record;
+
+  DeviceRecord records[VAR];
+
+  DEFINE_SIZE_MIN (8);
+
```
Comment (behdad) [inline src/hb-ot-hdmx-table.hh]: Remove extra lines. End of the struct should look like others, with public: and protected: and comments and ...

File: src/hb-ot-hdmx-table.hh [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+#define HB_OT_TAG_hdmx HB_TAG('h','d','m','x')
+
+
+struct DeviceRecord
+{
+  HBUINT8 pixel_size;
+  HBUINT8 max_width;
+  HBUINT8 widths[VAR];
+};
```
Comment (behdad) [inline src/hb-ot-hdmx-table.hh]: This needs DEFINE_SIZE_STATIC

Comment (behdad) [reply src/hb-ot-hdmx-table.hh]: and sanitize.

File: src/hb-ot-hdmx-table.hh [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  {
+    if (unlikely (i >= num_records)) return Null(DeviceRecord);
+    return StructAtOffset<DeviceRecord> (this, min_size + i * size_device_record);
+  }
+
+  inline bool sanitize (hb_sanitize_context_t *c) const
+  {
+    TRACE_SANITIZE (this);
+    return_trace (likely (c->check_struct (this)
+                          && version == 0));
```
Comment (behdad) [inline src/hb-ot-hdmx-table.hh]: This is wrong. Should check the array.

**Comment by garretrieger** [comment]
Fonts subsetted via harfbuzz were failing to pass ots due to hdmx being invalid.

File: src/hb-ot-hdmx-table.hh (hunk 1 above)
Comment (garretrieger) [reply src/hb-ot-hdmx-table.hh]: Done.

File: src/hb-ot-hdmx-table.hh (hunk 2 above)
Comment (garretrieger) [reply src/hb-ot-hdmx-table.hh]: Done.

File: src/hb-ot-hdmx-table.hh (hunk 3 above)
Comment (garretrieger) [reply src/hb-ot-hdmx-table.hh]: Done.

**Pull request merged**